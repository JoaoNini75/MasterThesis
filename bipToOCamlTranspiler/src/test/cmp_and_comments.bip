let nonneg (arg1) : bool = begin
	arg1 > -1 (* this is a comment *)
end
(*@ ensures true *)

(* another comment *)

let cmp1 () : bool (*3rd comment*) = begin
	not true == false
end
(*@ ensures true *)

let cmp2 (b1, b2) : bool = begin
	(b1 && b2) == (* nested comment: outer (* inner *) *) (b1 || b2)
end
(*@ ensures true *)

(* let x () = begin 4 end
this is code... or not *)
