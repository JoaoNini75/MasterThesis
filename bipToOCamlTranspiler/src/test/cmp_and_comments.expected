let nonneg (arg1) : bool =
  arg1 > -1
(*@ ensures true *)

let cmp1 () : bool =
  (not true) = true
(*@ ensures true *)

let cmp2 (b1) (b2) : bool =
  (b1 && b2) = (b1 || b2)
(*@ ensures true *)

